share/dkg/protocol.go

```diff
@@ -439,3 +409,80 @@ type OptionResult struct {
 	Result *Result
 	Error  error
 }
+
+type packet interface {
+	Hash() []byte
+	Index() Index
+}
+
+var _ packet = (*DealBundle)(nil)
+var _ packet = (*ResponseBundle)(nil)
+var _ packet = (*JustificationBundle)(nil)
+
+type set struct {
+	vals map[Index]packet
+	bad  []Index
+}
+
+func newSet() *set {
+	return &set{
+		vals: make(map[Index]packet),
+	}
+}
+
+func (s *set) Push(p packet) {
+	hash := p.Hash()
+	idx := p.Index()
+	if s.isBad(idx) {
+		// already misbehaved before
+		return
+	}
+	prev, present := s.vals[idx]
+	if present {
+		if !bytes.Equal(prev.Hash(), hash) {
```

Comment:
so if i broadcast an extra deal / response that you come to think is from someone else, i can knock out their legitimate message?

Reply:
That willl only happen if you dont use a signature scheme; otherwise, you are not supposed to be able to create a message looking like it comes from someone else. 
In fact, i think I should disallow *not* using signatures. Will do that now in another PR.